### PR TITLE
Feature/bngp 2175 email address correct

### PR DESCRIPTION
**Pull request by usman-j-shaikh**
This PR includes changes about email address confirmation.
- Email address validator
- Confirm email address page for DJ
- Modified helper method

### CODE REVIEW

File: packages/webapp/src/views/developer/details-email-confirm.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+{% block formContent %}
+<div class="govuk-body">
+    {% if err %}
+        {% include "errorbanner.html" %}
+    {% endif %}
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            {% set emailHtml %}
+                {% if errorMessage %}
```
Comment (teddmason) [inline packages/webapp/src/views/developer/details-email-confirm.html:14]: Don't think it needs the if statement here just to set the errorMessage, errorMessage could just be set as an empty object/array if not in error

File: packages/webapp/src/views/developer-form-layout.html
Comment (teddmason) [inline packages/webapp/src/views/developer-form-layout.html:1]: How come you need this file? looks to be nearly the same as the current form-layout.html file?

File: packages/webapp/src/utils/constants.js [hunk 2]
```diff
@@ -171,6 +171,8 @@ const DEVELOPER_FULL_NAME = 'developer-fullname'
 const DEVELOPER_REFERER = 'developer-referer'
 const DEVELOPER_DETAILS_EMAIL = 'developer/details-email'
 const DEVELOPER_EMAIL_VALUE = 'developer-email-value'
+const DEVELOPER_DETAILS_EMAIL_CONFIRM = 'developer/details-email-confirm'
+const DEVELOPER_CONFIRM_EMAIL = 'developer-yes'
```
Comment (teddmason) [inline packages/webapp/src/utils/constants.js:175]: This redis key needs a bit more description probably something like developer-confirm-email, developer-yes doesn't sound right?

**Review by teddmason — CHANGES_REQUESTED** [review]

File: packages/webapp/src/views/developer-form-layout.html
Comment (usman-j-shaikh) [reply packages/webapp/src/views/developer-form-layout.html:1]: We have different page title, and to define this title each and every page not sound good
And I think more changes will come in future which will slightly differ than LOJ
So I want to centralize this work using this file

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]

**Review by pwadmore-ea — APPROVED** [review]

**Review by teddmason — APPROVED** [review]

**Pull request merged**